x/mongo/driver/operation_exhaust.go

```diff
@@ -19,7 +19,7 @@ func (op Operation) ExecuteExhaust(ctx context.Context, conn StreamerConnection,
 	}
 
 	scratch = scratch[:0]
-	res, err := op.readWireMessage(ctx, conn, scratch)
+	res, _, err := op.readWireMessage(ctx, conn, scratch)
```

Comment:
Optional: Consider removing the `scratch` parameter from `ExecuteExhaust` as well and pulling a buffer from the pool for this function as well.

Reply:
`ExecuteExhaust` is only used in tests now in my understanding. So is it worth using a pool here? Correct me if I am wrong.

Reply:
`ExecuteExhaust` is used for the "streaming hello" [here](https://github.com/mongodb/mongo-go-driver/blob/f301e5d79010955a64026a264704145febcc6cfc/x/mongo/driver/operation/hello.go#L202), which is part of the server heartbeat logic. It's not expected to send/receive lots of data, so the memory use difference would be small. However, the `scratch` parameter is totally unused, so should be safe to remove.